config/sprayproxy.yaml

```diff
@@ -74,7 +93,21 @@ spec:
   selector:
     app.kubernetes.io/name: sprayproxy
   ports:
-    - name: https
+  - name: https
+    protocol: TCP
+    port: 8443
+    targetPort: https
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kube-rbac-proxy
```

Comment:
There is an existing "sprayproxy-service" - this should be updated to point to port 9443 instead of 8080. The route for the proxy goes through `sprayproxy-service.

Reply:
ok make sense 👍 

I have a question here. I found the existing route has [edge termination](https://github.com/redhat-appstudio/sprayproxy/blob/main/config/route.yaml#L13) which terminates the request with error
```
Client sent an HTTP request to an HTTPS server.
```

So I changed the [tls termination to passthrough](https://github.com/redhat-appstudio/sprayproxy/pull/28/commits/618e3c782b0661d819f3750e76177393af86c72f#diff-70d49e4e04db9e67659ff222c2d3bec6517227bf1d50ca97f434925d18e8fc10R11) and it works fine. However I would really like to hear from you if we can figure out something with edge termination as well.

Reply:
Good catch! Edge termination won't be an option because kube-rbac-proxy serves its data over HTTPS only. When a route uses edge termination, the thing doing the routing (HAProxy) forwards the request to the backend service using HTTP.

Passthrough termination won't be an option either because it invalidates the HSTS header that @AndrienkoAleksandr just added: https://docs.openshift.com/container-platform/4.11/networking/routes/route-configuration.html#nw-enabling-hsts_route-configuration. Passthrough just forwards the request directly using HTTPS. I'm frankly surprised this works and doesn't result in all sorts of "self signed certificate" errors.

That leaves us with "renecrypt" - which should work since as you suggest passthrough works fine.

Reply:
> That leaves us with "renecrypt" - which should work since as you suggest passthrough works fine.

@adambkaplan Sadly its not working out 🙁 Although passthrough does meets the requirement.

Reply:
In `re-encrypt` the active SSL will be terminated, and then re-encrypted using a different pair of cert and key. In this case since its an Openshift platform, you can get a certificate generated using service ca and provide it to the route as well as the kube-rbac proxy.

Reply:
add this annotation in the service.. 
`service.beta.openshift.io/serving-cert-secret-name: sprayproxy-tls`

  and the following arguments in the kube-rbac specs, after mounting the tls volume in /etc/tls.
```
- '--tls-cert-file=/etc/tls/tls.crt'
- '--tls-private-key-file=/etc/tls/tls.key'
```

Reply:
Thanks @sayan-biswas for helping out here. I have pushed the required changes 🙂